src/components/SavedNews.js

```diff
@@ -1,15 +1,43 @@
-import React, { useState } from 'react';
+import React, { useEffect } from 'react';
 import SavedNewsHeader from './SavedNewsHeader';
 import Footer from './Footer';
 import Search from './Search';
 import Temp from './Temp';
 import FooterMobile from './FooterMobile';
+import MainApi from '../utils/MainApi';
+import { useState } from 'react/cjs/react.development';
 
 function SavedNews(props) {
+  const [cards, setCards] = useState([]);
+  const [numberOfArticles, setNumberOfArticles] = useState(cards.length);
+
+  useEffect(() => {
+    setNumberOfArticles(cards.length);
+  }, [cards.length]);
+
+  useEffect(() => {
+    MainApi.getSavedArticles(props.jwt)
+      .then((res) => {
+        let body = res;
```

Comment:
If a variable is declared using `let`, its value must change somewhere.
If a variable's value doesn’t change, it is declared using `const`.

All your variables should be declared using `const`